src/include/daos_obj.h

```diff
@@ -564,6 +564,21 @@ daos_obj_generate_oid(daos_handle_t coh, daos_obj_id_t *oid,
 		      enum daos_otype_t type, daos_oclass_id_t cid,
 		      daos_oclass_hints_t hints, uint32_t args);
 
+
+/**
+ * This function, if called 2^32 times will set oid->hi to every unique 32-bit
```

Comment:
I'm confused by the first sentence of this method. Are oids 32 bit integers which means there are 2^32 possible combinations? Is this function just getting us a unique `hi` every time it is called? When does it get called?

Reply:
The oid has 2 64-bit integers, lo and hi.   The lower 96 bits (all of lo and the lo 32-bits of hi) are supposed to be unique within a container for a given object.   DAOS provides a method that allocates a globally unique lo oid upon request for the container.   The user, in this case DFS, then has 2^32 unique oids it can allocate to produce a globally unique oid within the container.   When it runs out, it will request a new lo (it probably won't run out as that would require the process to create 2^32 + 1 objects).

This function is just a helper function to cycle through the range of available oids.

Reply:
The upper 32-bits of hi are used by DAOS to encode things like the object class